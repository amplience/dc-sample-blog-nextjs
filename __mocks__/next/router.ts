export const mockRouterOn = jest.fn();
export const Router = { events: { on: mockRouterOn } };
export const useRouter = (): { asPath: string; query: unknown } => ({ asPath: 'mocked-path', query: {} });
