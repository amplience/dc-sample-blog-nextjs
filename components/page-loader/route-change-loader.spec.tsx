/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import RouteChangeLoader from './route-change-loader';
import { mockRouterOn } from 'next/router';
import { shallow } from 'enzyme';

describe('RouteChangeLoader', () => {
  test('renders route change loader when loading', async () => {
    const wrapper = shallow(<RouteChangeLoader />);
    wrapper.setState({ isLoading: true });
    expect(wrapper.html()).toMatchSnapshot();
  });
  test('renders route change loader when not loading', async () => {
    const wrapper = shallow(<RouteChangeLoader />);
    wrapper.setState({ isLoading: false });
    expect(wrapper.html()).toMatchSnapshot();
  });
  test('renders route change loader with correct event listeners', async () => {
    await renderer.create(<RouteChangeLoader />);
    expect(mockRouterOn.mock.calls[0][0]).toEqual('routeChangeStart');
    expect(mockRouterOn.mock.calls[1][0]).toEqual('routeChangeComplete');
    expect(mockRouterOn.mock.calls[2][0]).toEqual('routeChangeError');
  });
  test('renders route change loader with routeChangeStart event listener and execute function', async () => {
    let routeChangeStartCallback;
    const mockRouteChangeStartFn = (event, callback) => {
      if (event === 'routeChangeStart') {
        routeChangeStartCallback = callback;
      }
    };
    mockRouterOn.mockImplementation(mockRouteChangeStartFn);
    const wrapper = shallow(<RouteChangeLoader />);
    // forcing isLoading to true so we know it is changed by the callback
    wrapper.setState({ isLoading: false });
    routeChangeStartCallback();
    expect(wrapper.state('isLoading')).toBe(true);
  });

  test('renders route change loader with routeChangeComplete event listener and execute function', async () => {
    let routeChangeCompleteCallback;
    const mockRouteChangeComplete = (event, callback) => {
      if (event === 'routeChangeComplete') {
        routeChangeCompleteCallback = callback;
      }
    };
    mockRouterOn.mockImplementation(mockRouteChangeComplete);
    const wrapper = shallow(<RouteChangeLoader />);
    // forcing isLoading to true so we know it is changed by the callback
    wrapper.setState({ isLoading: true });
    routeChangeCompleteCallback();
    expect(wrapper.state('isLoading')).toBe(false);
  });

  test('renders route change loader with routeChangeError event listener and execute function', async () => {
    let routeChangeErrorCallback;
    const mockRouteChangeErrorComplete = (event, callback) => {
      if (event === 'routeChangeError') {
        routeChangeErrorCallback = callback;
      }
    };
    mockRouterOn.mockImplementation(mockRouteChangeErrorComplete);
    const wrapper = shallow(<RouteChangeLoader />);
    // forcing isLoading to true so we know it is changed by the callback
    wrapper.setState({ isLoading: true });
    routeChangeErrorCallback();
    expect(wrapper.state('isLoading')).toBe(false);
  });
});
