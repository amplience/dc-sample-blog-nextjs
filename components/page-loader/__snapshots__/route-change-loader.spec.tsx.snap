// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteChangeLoader renders route change loader when loading 1`] = `"<div class=\\"jsx-1481206946 page-loading\\"><div class=\\"jsx-1481206946 spinner\\"></div></div>"`;

exports[`RouteChangeLoader renders route change loader when not loading 1`] = `""`;
